src/main/kotlin/ecommerce/service/PaymentService.kt

```diff
@@ -0,0 +1,69 @@
+package ecommerce.service
+
+import ecommerce.client.StripeClient
+import ecommerce.dto.OrderRequest
+import ecommerce.dto.PaymentResponse
+import ecommerce.enum.PaymentStatus
+import ecommerce.exception.StripePaymentException
+import ecommerce.model.Payment
+import ecommerce.repository.PaymentJpaRepository
+import ecommerce.repository.getByPaymentIntentIdOrThrow
+import jakarta.transaction.Transactional
```

Comment:
I think the annotation that you were trying to import was `org.springframework.transaction.annotation.Transactional`, not `jakarta.transaction.Transactional`!

[We went over this before](https://github.com/herotechcourse/spring-ecommerce-order/pull/7#discussion_r2249306942), so I hope we acknowledge their difference and import Spring's annotation. 🙂